### ISSUE TITLE
TextSecure randomly pops up

### ISSUE THREAD

**Issue by denali1** [WhisperSystems/TextSecure#2239]
I can't nail down why TS does this.  Sometimes, when I unlock the phone or exit an app, TextSecure will pop up asking for my password.  When it does this, my finger is no where near the TS icon.  Log follows:

```
Device : LGE Nexus 5 (hammerhead)
Android: LRX22C
OS Host: vpbs16.mtv.corp.google.com
App    : TextSecure 2.3.3

--------- beginning of system
W/ResourcesManager(18639): Asset path '/system/framework/android.test.runner.jar' does not exist or contains no resources.
--------- beginning of main
W/PartProvider(18639): Got master secret: null
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
D/OpenGLRenderer(18639): Render dirty regions requested: true
D/Atlas   (18639): Validating map...
I/Adreno-EGL(18639): <qeglDrvAPI_eglInitialize:410>: QUALCOMM Build: 10/24/14, 167c270, I68fa98814b
I/OpenGLRenderer(18639): Initialized EGL, version 1.4
D/OpenGLRenderer(18639): Enabling debug mode 0
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
I/Choreographer(18639): Skipped 31 frames!  The application may be doing too much work on its main thread.
W/MmsCommunication(18639): Getting MMSC params for apn null
W/KeyCachingService(18639): Incrementing activity count...
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/KeyCachingService(18639): Decrementing activity count...
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsCommunication(18639): Getting MMSC params for apn null
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsCommunication(18639): Getting MMSC params for apn null
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsCommunication(18639): Getting MMSC params for apn null
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsCommunication(18639): Getting MMSC params for apn null
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 2) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 2) ORDER BY date_received ASC)
W/PhoneNumberFormatter(18639): Got local CC: US
W/MmsCommunication(18639): Getting MMSC params for apn null
W/TelephonyUtil(18639): Choosing MCC+MNC info from TelephonyManager.getSimOperator()
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/ApnDatabase(18639): Choosing locally-overridden MMS settings
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
W/DirectoryRefreshListener(18639): Scheduling for: 1418952253439
W/ResourcesManager(18639): Asset path '/system/framework/com.google.android.media.effects.jar' does not exist or contains no resources.
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/ConversationListActivity(18639): onDestroy...
W/IInputConnectionWrapper(18639): showStatusIcon on inactive InputConnection
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/service (18639): Broadcasting new secret...
W/KeyCachingService(18639): Starting timeout: 3600000
W/MmsSmsDatabase(18639): Executing query: SELECT _id, body, read, type, address, address_device_id, subject, thread_id, status, date_sent, date_received, m_type, msg_box, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (read = 0) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (read = 0) ORDER BY date_received ASC)
I/art     (18639): Explicit concurrent mark sweep GC freed 7057(473KB) AllocSpace objects, 13(549KB) LOS objects, 40% free, 22MB/37MB, paused 700us total 35.682ms
W/SendReceiveService(18639): Got a MasterSecret broadcast...
W/PartProvider(18639): Got master secret: org.thoughtcrime.securesms.crypto.MasterSecret@17d32a28
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/KeyCachingService(18639): Starting timeout: 3600000
W/DirectoryRefreshListener(18639): Scheduling for: 1418952253439
W/ResourcesManager(18639): Asset path '/system/framework/com.google.android.media.effects.jar' does not exist or contains no resources.
W/KeyCachingService(18639): Incrementing activity count...
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/RecipientProvider(18639): Cache miss [ASYNC]!
W/KeyCachingService(18639): Decrementing activity count...
W/KeyCachingService(18639): Starting timeout: 3600000
W/Resources(18639): Converting to string: TypedValue{t=0x10/d=0x1f4 a=-1}
W/Resources(18639): Converting to string: TypedValue{t=0x10/d=0x1f4 a=-1}
E/MediaPlayer(18639): Should have subtitle controller already set
D/Ringtone(18639): Successfully created local player
E/MediaPlayer(18639): Should have subtitle controller already set
W/KeyCachingService(18639): Incrementing activity count...
W/KeyCachingService(18639): Decrementing activity count...
W/KeyCachingService(18639): Starting timeout: 3600000
```

**Comment by jelle-e** [comment]
Yep, this behaviour began after upgrading to lollipop on my nexus5. Same issues as above.

**Comment by irv** [comment]
getting something similar after upgrade to lollipop on my nexus 5 too. seems like the service is getting killed randomly.

**Comment by McLoo** [comment]
confirmed for CM12
When the timeout locks TS it pops up to say: *I'm locked now, unlock me* :)

**Comment by Mihahn** [comment]
I can confirm this! Experiencing the same issue on my Nexus 4 (Android 5.0.1 / TextSecure 2.3.3) locked with a passphrase, the timeout interval is set to 60 minutes.

Another thing I've noticed is that when I remove TextSecure from the multitask menu to kill it, this menu gets closed and I'm back on my homescreen. Should I open up a new issue for this? I guess this is a bug on Lollipop...

**Comment by McLoo** [comment]
@Mihahn the task close issue is a known issue. Can't do much about it. Just don't force close TS😊

**Comment by McLoo** [comment]
Popup also happens when locking from the notification

**Comment by irv** [comment]
seems likely it's this android bug? https://code.google.com/p/android/issues/detail?id=79729

**Comment by deusinvictus** [comment]
@irv I disagree that is related to the Lollipop memory leak.  I am also experiencing the memory leak on my phone, but apps don't start crashing from the memory leak until several days (typically 3-5) after a reboot.  Where as this bug with Text Secure happens consistantly even only a couple of hours after a reboot.  Also the memory leak causes the "Sysmtem" memory to be very high when looking in the system settings and several times I've looked the "System" memory was still quite low when I've had TextSecure pull itself to the forefront and asked to be unlocked.

**Comment by tavianator** [comment]
It's an Android bug, but not the one that @irv linked:

https://code.google.com/p/android/issues/detail?id=63570
https://code.google.com/p/android/issues/detail?id=71067

**Comment by deusinvictus** [comment]
@tavianator I'm not sure that's it either, because I've had TS pop up without ever hitting the back button from within another app.  I've been looking at me screen reading a web page or PDF, etc not even touching the screen and it pulls itself to the front.

**Comment by gabrielbauman** [comment]
Nexus 5, Lollipop; I get this at least three times a day. App Protection is on. It doesn't matter what I am doing - I might be typing an email or browsing reddit or playing a game and suddenly TextSecure is in the foreground asking for my passphrase.

**Comment by gabrielbauman** [comment]
Just happened to me again. https://gist.github.com/anonymous/0cff7221933d2134fc17

**Comment by gabrielbauman** [comment]
And again: https://gist.github.com/anonymous/b5f68da0eff07af6261e

**Comment by gabrielbauman** [comment]
And again,  this time interrupted me mid email: https://gist.github.com/d3d742885b8ff56b0165

**Comment by gabrielbauman** [comment]
Still an issue in 2.6.4 from the Play Store.

With a ten minute lock timeout set in TextSecure, I am interrupted by the unlock activity after ten minutes, no matter what I am doing on the phone. It pushes its way to the top and I have to dismiss it, sometimes twice.
